samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+ * Voter: This is a console application that simulates casting of votes. It is hosted outside of the Service Fabric cluster. 
+ * CandidateVoteCount: Service Fabric service with the logic to count votes by candidate while the votes come in. It also instructs the `ZipCodeVoteCount` endpoint to track votes by zip code. It will report the intermediate results as well as the final results when the election is closed.
+ * ZipCodeVoteCount: Service Fabric service with the logic to count the votes by zip code in the background. It will report the results when the allowed counting period is over.
+ * ServiceFabricRouting: [Service Fabric application](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-application-model) containing description of services it will run when deployed into Service Fabric cluster.
+
+
+## Cluster partitioning
+
+The `CandidateVoteCount` is a [stateful service](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-introduction) that uses a `NamedPartition` [partitioning scheme](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-concepts-partitioning). Each candidate has its own partition, so there is one called "John" and another called "Abby". 
+
+The `ZipCodeVoteCount` is a stateful service that uses a `UniformInt64Partition` partitioning scheme with the `PartitionCount` set to 3, the `LowKey` set to 0 and the `HighKey` set to 99000. This configuration ensure that the partition is split into 3 well known ranges (0 -> 32999), (33000 -> 65999) , (66000 -> 98999).
+
+
+## Routing
+
+The default NServiceBus routing approach cannot be used as-is with Service Fabric stateful services. Stateful services assume business data partitioning. A message must be delivered to a specific replica (instance) of a stateful that can handle  the message data. E.g. for each `PlaceVote` message associated with a casted vote, the message should be routed to the partition associated with voted candidate, "John" or "Abby". Similar to that, to count the votes per zip code the `TrackZipCode` message needs to end up on the partition that is responsible for the range to which the zip code belongs.
+
+Example:
+
+- Vote for John; casted in 88701; should result in a `PlaceVote` message routed to named partition "John", followed by a `TrackZipCode` message routed to range partition (66000 -> 98999).
+- Vote for Abby; casted in 36458; should result in a `PlaceVote` message routed to named partition "Abby", followed by a `TrackZipCode` message routed to range partition (33000 -> 65999).
+- Vote for John; casted in 12789; should result in a `PlaceVote` message routed to named partition "John", followed by a `TrackZipCode` message routed to range partition (0 -> 32999).
+
+and so on.
+
+Partition aware behavior is achieved by combining NServiceBus built-in sender side distribution feature and a few modifications to the processing pipeline.
```

Comment:
Instance-mapping file if for MSMQ only. We are leveraging the core SSD behavior.